# Proses Gambar Sumber RAW Output A (modified)

In [21]:
import pandas as pd
df_hasil_a = pd.read_csv('csv/pernah.csv')
df_hasil_a.head(3)

filename
0     10151.png
1  112_img_.png
2  121_img_.png

In [8]:
import pandas as pd
df_hasil_a = pd.read_csv('hasil/2024_haofeng_hu_a.csv')
df_hasil_a.head(3)

id      filename      cast
0   0  100_img_.png    bluish
1   1     10139.png  greenish
2   2     10151.png  greenish

In [16]:
# find which rows have "no-color-cast" in the "cast" column
# df_hasil_a['cast']=='no-color-cast'
# filter dataframe based on condition
# df_hasil_a[df_hasil_a['cast']=='no-color-cast']
# verify which hasil are in the "cast" column with value_counts()
print(df_hasil_a[df_hasil_a['cast']=='no-color-cast'].value_counts('cast'))
print(df_hasil_a[df_hasil_a['cast']=='bluish'].value_counts('cast'))
print(df_hasil_a[df_hasil_a['cast']=='greenish'].value_counts('cast'))
print(df_hasil_a[df_hasil_a['cast']=='yellowish'].value_counts('cast'))

cast
no-color-cast    219
Name: count, dtype: int64
cast
bluish    236
Name: count, dtype: int64
cast
greenish    371
Name: count, dtype: int64
cast
yellowish    64
Name: count, dtype: int64


In [19]:
import pandas as pd
df_hasil_b = pd.read_csv('hasil/2024_haofeng_hu_b.csv')
df_hasil_b.head(3)
print(df_hasil_b[df_hasil_b['cast']=='no-color-cast'].value_counts('cast'))
print(df_hasil_b[df_hasil_b['cast']=='bluish'].value_counts('cast'))
print(df_hasil_b[df_hasil_b['cast']=='greenish'].value_counts('cast'))
print(df_hasil_b[df_hasil_b['cast']=='yellowish'].value_counts('cast'))

cast
no-color-cast    732
Name: count, dtype: int64
cast
bluish    73
Name: count, dtype: int64
cast
greenish    32
Name: count, dtype: int64
cast
yellowish    53
Name: count, dtype: int64


In [2]:
import os,csv
# 1) Baca CSV untuk skip file yang sudah diproses
input_csv = "2024_haofeng_hu_a.csv"
done = set()
if os.path.exists(input_csv):
    try:
        with open(input_csv, "r", newline="", encoding="utf-8") as f:
            reader = csv.reader(f)
            header = next(reader, None)
            for row in reader:
                if row:
                    done.add(os.path.abspath(row[0]))
    except Exception as e:
        print(f"[PERINGATAN] Gagal baca CSV: {e}, lanjut tanpa resume.")
        done = set()

# 2) Tentukan sisa file
all_files = []
limit = None
remaining = [p for p in all_files if p not in done]
if limit is not None:
    remaining = remaining[:max(0, limit)]

print(done)

set()


102_img_.png greenish -> yellowish 1.5316730482492649
104_img_.png bluish -> bluish 1.2097321665161656
107_img_.png bluish -> bluish 1.2932880980142356
10_img_.png bluish -> bluish 1.300357854702698
108_img_.png greenish-> greenish 3.510571732480775
10945.png bluish-> greenish 1.0407622980306688
111_img_.png bluish-> bluish 1.1859838456442136
113_img_.png greenish-> yellowish 1.003210833280044
114_img_.png bluish-> bluish 1.1891120339449206
115_img_.png bluish-> bluish 1.5829973269807742
120_img_.png bluish-> bluish 1.016073242153681
12336.png no-color-cast-> greenish 1.2415023879571245
12433.png greenish-> yellowish 1.6032485068771687
12445.png bluish-> yellowish 2.008122341539302
12900.png greenish-> yellowish 2.172484898001756
131_img_.png bluish-> bluish 1.3387502447838295
138_img_.png bluish-> bluish 1.182464706090961
145_img_.png bluish-> greenish 1.6352173290024223
15094.png greenish-> greenish 1.0127638362657934
1742.png greenish-> yellowish 1.5919935234413785
18_img_.png greenish-> yellowish 1.075050964656185
186_img_.png bluish-> bluish 1.1582241740771666
189_img_.png greenish-> greenish 1.6245534725721837
202_img_.png yellowish-> yellowish 1.963733117748857
208_img_.png greenish-> yellowish 1.157954396668117
229_img_.png bluish-> yellowish 2.171390434249527
272_img_.png no-color-cast-> greenish 1.0179095067262882
274_img_.png greenish-> greenish 1.5092736805163658
276_img_.png no-color-cast-> greenish 1.0869144440137284
296_img_.png greenish-> greenish 1.199391382743877
10139.png greenish-> no-color-cast 0.6653257251842517
105_img_.png greenish-> no-color-cast 0.3276532609866705
116_img_.png greenish-> no-color-cast 0.5152270270089745
12299.png no-color-cast-> no-color-cast 0.8919064820498562
137_img_.png bluish-> no-color-cast 0.4655331069674995
1407.png no-color-cast-> no-color-cast 0.4571113050011408
143_img_.png greenish-> no-color-cast 0.3964636664049659
15045.png greenish-> no-color-cast 0.4234045229225008
151_img_.png bluish-> no-color-cast 0.915898779928032
158_img_.png no-color-cast-> no-color-cast 0.30554817308465326